include/particles/particle.tcc

```diff
@@ -852,10 +855,16 @@ void mpm::Particle<Tdim>::apply_particle_velocity_constraints(unsigned dir,
   this->velocity_(dir) = velocity;
 }
 
+//! Return particle scalar data
+template <unsigned Tdim>
+double mpm::Particle<Tdim>::scalar_data(const std::string& property) {
+  return this->scalar_properties_.at(property)();
```

Comment:
We need to return a `nan` if we can't find the scalar property, otherwise if we have multiple materials in the mesh and we write vtk file, where one material has a property that the other doesn't, then the code will crash. We should do this for state variables too.

Reply:
I agree with you. I will change the one in `state_vars` too. How about `tensor_properties` though? We might have `liquid_velocities` or `permeability_` for instance, they are dimensional vector.

Reply:
We return nans for scalar, vector and tensorial properties. Do we not have vector_properties anymore?

Reply:
Nope, it was combined before in your `tensor_data()` function as `properties_`. I changed it to `tensor_properties_` after having `scalar_properties_`. I can separate them to make it coherence and clearer with the VariableType naming. Will that be good?

Reply:
I think having `vector_properties_` is good. Wouldn't not having that impact how they are written in vtk? as in would they still be written as vtkvectors? Either way, it's good to have a separate `vector_properties`

Reply:
OK. I'm happy not needing a separate `vector_properties` in here.